test/test-listing-guni.spec.ts

```diff
@@ -0,0 +1,233 @@
+
+    // impersonating holders
+
+    whale = ethers.provider.getSigner(AAVE_WHALE);
+    guniHolder = ethers.provider.getSigner(GUNI_HOLDER);
+    daiHolder = ethers.provider.getSigner(DAI_HOLDER);
+
+    //getting main entry point contracts
+    gov = (await ethers.getContractAt(
+      'IAaveGovernanceV2',
+      AAVE_GOVERNANCE_V2,
+      proposer
+    )) as IAaveGovernanceV2;
+    pool = (await ethers.getContractAt(
+      'ILendingPool',
+      AAVE_LENDING_POOL,
+      proposer
+    )) as ILendingPool;
+
+    // getting tokens used for tests
+    aave = (await ethers.getContractAt('IERC20', AAVE_TOKEN, whale)) as IERC20;
+    dai = (await ethers.getContractAt('IERC20', DAI_TOKEN, daiHolder)) as IERC20;
+    guni = (await ethers.getContractAt('IERC20', GUNI1, guniHolder)) as IERC20;
+    oracle = (await ethers.getContractAt('IAaveOracle', AAVE_ORACLE)) as IAaveOracle;
+    decimalMultiplier = BigNumber.from('10').pow(BigNumber.from("18"));
+
+    // Give aave to proposer
+    await (
+      await aave.transfer(
+        proposer.address,
+        (await aave.balanceOf(AAVE_WHALE)).sub(parseEther('10000'))
+      )
+    ).wait();
+
+    // giving just a bit of Dai to GUNI holder to pay for interest later
+    await (await dai.transfer(GUNI_HOLDER, parseEther('10'))).wait();
+    
+    // give GUNI to proposer
+    const gUniBalance = await guni.balanceOf(GUNI_HOLDER);
+    await (
+      await guni.transfer(
+        proposer.address,
+        gUniBalance.div(BigNumber.from("5"))
+      )
+    ).wait();
+
+
+    // deploying the payload
+    await rawBRE.ethers.provider.send('evm_mine', [0]);
+
+    await rawBRE.deployments.deploy('AssetListingGUni', {
+      from: proposer.address,
+      gasLimit: 4000000,
+      gasPrice: BigNumber.from('75000000000'),
+      args: [],
+    });
+
+
+    proposal = await gov.getProposalsCount();
+
+    await rawBRE.run('create:proposal-new-asset:guni');
+
+    const payloadAddress = (
+      await rawBRE.deployments.get('AssetListingGUni')
+    ).address;
+    console.log(payloadAddress)
+
+    // voting, queuing proposals
+    await rawBRE.ethers.provider.send('evm_mine', [0]);
+
+    await (await gov.submitVote(proposal, true)).wait();
+    await advanceBlockTo((await latestBlock()) + VOTING_DURATION + 1);
+    await (await gov.queue(proposal)).wait();
+    let proposalState = await gov.getProposalState(proposal);
+    expect(proposalState).to.be.equal(5);
+    await increaseTime(86400 + 10);
+    snapshotId = await evmSnapshot();
+  });
+
+  it('Should list correctly an asset: borrow off, collateral on, stable borrow off', async () => {
+    await (await gov.execute(proposal)).wait();
+    
+    const proposalState = await gov.getProposalState(proposal);
+    expect(proposalState).to.be.equal(7);
+    const {
+      configuration: { data },
+      aTokenAddress,
+      stableDebtTokenAddress,
+      variableDebtTokenAddress,
+    } = await pool.getReserveData(GUNI1);
```

Comment:
could you add tests for all the assets the proposal is listing?

Reply:
yes, thats true - added